src/main/java/emissary/core/channels/ImmutableChannel.java

```diff
@@ -0,0 +1,72 @@
+package emissary.core.channels;
+
+import org.apache.commons.lang3.Validate;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.NonWritableChannelException;
+import java.nio.channels.SeekableByteChannel;
+
+/**
+ * Decorator class that wraps an existing {@link SeekableByteChannel} instance and makes it immutable
+ * 
+ * @param <T> Type of SeekableByteChannel to wrap
+ */
+public final class ImmutableChannel<T extends SeekableByteChannel> implements SeekableByteChannel {
```

Comment:
This file demonstrates one of the cases for which I broke some of the existing encapsulation by taking a previously `private static` inner class and making it public class.